docarray/array/storage/memory/getsetdel.py

```diff
@@ -53,28 +28,17 @@ def _set_doc_value_pairs(
         for _d, _v in zip(docs, values):
             _d._data = _v._data
 
-    def _set_doc_attr_by_offset(self, offset: int, attr: str, value: Any):
-        setattr(self._data[offset], attr, value)
-
-    def _set_doc_attr_by_id(self, _id: str, attr: str, value: Any):
-        if attr == 'id':
-            old_idx = self._id2offset.pop(_id)
-            setattr(self._data[old_idx], attr, value)
-            self._id2offset[value] = old_idx
-        else:
-            setattr(self._data[self._id2offset[_id]], attr, value)
-
-    def _get_doc_by_offset(self, offset: int) -> 'Document':
-        return self._data[offset]
-
     def _get_doc_by_id(self, _id: str) -> 'Document':
-        return self._data[self._id2offset[_id]]
+        return self._data[_id]
 
-    def _get_docs_by_slice(self, _slice: slice) -> Iterable['Document']:
-        return self._data[_slice]
+    def _get_docs_by_ids(self, ids: Sequence[str]) -> Iterable['Document']:
+        return (self._data[_id] for _id in ids)
 
-    def _get_docs_by_offsets(self, offsets: Sequence[int]) -> Iterable['Document']:
-        return (self._data[t] for t in offsets)
+    def _clear_storage(self):
+        self._data.clear()
```

Comment:
Where the `offset2ids` is cleared?

Reply:
this method is only about clearinng the storage, not offset2ids
It is cleared in a parent class